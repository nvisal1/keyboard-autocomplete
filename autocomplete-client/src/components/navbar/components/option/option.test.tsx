import React from 'react';
import renderer from 'react-test-renderer';

import Option, { OptionProps } from './option';

describe('When the Option component is created', () => {
    describe('and the isSelected prop is set to true', () => {
        it('should render correctly', () => {
            const requiredProps = getRequiredProps(true);
            const currentSnapshot = renderer.create(<Option { ...requiredProps }></Option>).toJSON();
    
            expect(currentSnapshot).toMatchSnapshot();
        });
    });
    describe('and the isSelected prop is set to false', () => {
        it('should render correctly', () => {
            const requiredProps = getRequiredProps(false);
            const currentSnapshot = renderer.create(<Option { ...requiredProps }></Option>).toJSON();
    
            expect(currentSnapshot).toMatchSnapshot();
        });
    });
});

function getRequiredProps(isSelected: boolean): OptionProps {
   const handleClick = (_: string) => {}; 

   const text = 'test_option';

   return { handleClick, isSelected, text };
}