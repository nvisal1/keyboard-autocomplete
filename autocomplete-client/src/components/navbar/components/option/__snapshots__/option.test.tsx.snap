// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the Option component is created and the isSelected prop is set to false should render correctly 1`] = `
<div
  className="option-container__text"
  onClick={[Function]}
  style={
    Object {
      "color": "",
    }
  }
>
  test_option
</div>
`;

exports[`When the Option component is created and the isSelected prop is set to true should render correctly 1`] = `
<div
  className="option-container__text"
  onClick={[Function]}
  style={
    Object {
      "color": "",
    }
  }
>
  test_option
</div>
`;
